### PR TITLE
Little changes in CoAP doc

### PR DESCRIPTION
**Pull request by alashkin**

### CODE REVIEW

File: src/coap.c [hunk 1]
```diff
@@ -54,7 +52,7 @@ void ns_coap_free_options(struct ns_coap_message *cm) {
 
 /*
  * Adds new option to ns_coap_message structure.
- * Returns pointer to newly created options.
+ * Returns pointer to newly created option.
```
Comment (mmikulicic) [inline src/coap.c]: to _the_ newly created option

File: src/coap.c [hunk 2]
```diff
@@ -333,8 +331,8 @@ static char *coap_get_options(char* ptr, struct iobuf *io,
  * (behind its RFC)
  * Caller have to check results and
  * threat COAP_NOT_ENOUGH_DATA according to
- * underlaying protocol
- * in case of UDP COAP_NOT_ENOUGH_DATA means COAP_FORMAT_ERROR
+ * underlaying protocol:
```
Comment (mmikulicic) [inline src/coap.c]: s/underlaying/underlying/

File: src/coap.c [hunk 3]
```diff
@@ -492,7 +490,7 @@ static uint32_t coap_calculate_packet_size(struct ns_coap_message *cm,
 /*
  * Composes CoAP message from ns_coap_message structure.
  * Returns 0 on success or bit mask with wrong field.
- * Packet returned must be explicitly deleted by NS_FREE.
+ * Packet returned must be explicitly freed by iobuf_free.
```
Comment (mmikulicic) [inline src/coap.c]: Well not necessarily. The whole point of iobuffers is that they can be reused. I'd just drop this line; the behaviour is aligned with the fossa buffer management patterns.

**Comment by mmikulicic** [comment]
LGTM, minor comments

File: src/coap.c (hunk 1 above)
Comment (alashkin) [inline src/coap.c]: Done.

File: src/coap.c (hunk 2 above)
Comment (alashkin) [inline src/coap.c]: Done.

File: src/coap.c (hunk 3 above)
Comment (alashkin) [inline src/coap.c]: Done.

**Comment by alashkin** [comment]
PTAL

**Pull request merged**